bin/p2-preparer/main.go

```diff
@@ -43,6 +45,16 @@ func main() {
 
 	quitMainUpdate := make(chan struct{})
 	quitHookUpdate := make(chan struct{})
+	quitWatchHealth := make(chan struct{})
+	if preparerConfig.ConsulTokenPath != "" {
```

Comment:
The logic of loading the `ConsulTokenPath` should really happen in `setup.go` - this file is purposely quite lean to encourage modularity.